internal/restore/volumesnapshotrestore_action.go

```diff
@@ -71,19 +77,101 @@ func (p *VolumeSnapshotRestoreRestoreItemAction) Execute(input *velero.RestoreIt
+	err = vsrClient.Get(context.Background(), client.ObjectKey{Namespace: vsr.Namespace, Name: vsr.Name}, &vsr)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error fetching volumesnapshotrestore CR for suppyling operationID")
+	}
+
+	// operationID for our datamover usecase is VSR NamespacedName which will unique per operation
+	operationID = vsr.Namespace + "/" + vsr.Name
+
+	p.Log.Info("Returning from VolumeSnapshotRestoreRestoreItemActionV2")
+
 	// don't restore VSB
 	return &velero.RestoreItemActionExecuteOutput{
-		SkipRestore: true,
+		SkipRestore: true, OperationID: operationID,
 	}, nil
 }
+
+func (p *VolumeSnapshotRestoreRestoreItemActionV2) Progress(operationID string, restore *v1.Restore) (velero.OperationProgress, error) {
+	progress := velero.OperationProgress{}
+
+	// handle empty operationID case
+	if operationID == "" {
+		return progress, riav2.InvalidOperationIDError(operationID)
+	}
+
+	// fetch the VSR matching the operationID supplied, read its status and return progress of datamovement
+	vsrClient, err := util.GetVolumeSnapshotMoverClient()
+	vsr := datamoverv1alpha1.VolumeSnapshotRestore{}
+	if err != nil {
+		return progress, errors.Wrapf(err, "error getting volumesnapshotrestore client")
+	}
+
+	splitOperationID := strings.Split(operationID, "/")
+	if len(splitOperationID) != 2 {
+		return progress, riav2.InvalidOperationIDError(operationID)
+	}
+
+	VSRNamespace := splitOperationID[0]
+	VSRName := splitOperationID[1]
+
+	err = vsrClient.Get(context.Background(), client.ObjectKey{Namespace: VSRNamespace, Name: VSRName}, &vsr)
+	if err != nil {
+		return progress, errors.Wrapf(err, "error fetching volumesnapshotrestore CR for operationID: %s", operationID)
+	}
+
+	// update progress status via VSR phases
+	if vsr.Status.Phase != "" && vsr.Status.BatchingStatus != "" {
+
+		progressDescriptionPhase := string(vsr.Status.Phase)
+		progressDescriptionBatchingStatus := string(vsr.Status.BatchingStatus)
+		progress.Description = "Phase: " + progressDescriptionPhase + " BatchingStatus: " + progressDescriptionBatchingStatus
+
+		p.Log.Infof("current progress description is: %s", progress.Description)
+
+		if vsr.Status.Phase == datamoverv1alpha1.SnapMoverRestorePhaseCompleted {
+			progress.Completed = true
+		}
+
+		if vsr.Status.Phase == datamoverv1alpha1.SnapMoverRestorePhaseFailed {
+			progress.Err = "VolumeSnapshotRestore has a failed status"
+			progress.Completed = true
+		}
+	}
+
+	// update progress timestamps
+	if vsr.Status.StartTimestamp != nil {
+		progress.Started = vsr.Status.StartTimestamp.Time
+	}
+
+	// mark updated timestamp
+	progress.Updated = time.Now()
+
+	return progress, nil
+}
+
+func (p *VolumeSnapshotRestoreRestoreItemActionV2) Cancel(operationID string, restore *v1.Restore) error {
+	return nil
+}
+
+func (p *VolumeSnapshotRestoreRestoreItemActionV2) AreAdditionalItemsReady(additionalItems []velero.ResourceIdentifier, restore *v1.Restore) (bool, error) {
+	return true, nil
+}
```

Comment:
ehh these two funcs are empty and doesn't do anything with the params

Reply:
@kaovilai Yes, they're needed to implement the interface. If canceling the action is possible, then some implementation could be added.

Reply:
Then comment as not implemented/

Reply:
`// empty funcs to satisfy interface`